languages/Dockerfile.compile.j2

```diff
@@ -1,19 +1,27 @@
 FROM {{builder_image}} as builder
-USER 2222
+
 # Docker build commands don't resolve environment variables so need this to either be numeric or a build argument
-COPY --chown=2222:2222 algosource /opt/algorithm/
+COPY --chown=algo:algo algosource /opt/algorithm/
 
 {% if config.local_dependency_src_path is defined %}
-COPY --chown=2222:2222 {{config.local_dependency_src_path}} {{config.local_dependency_dest_path}}
+COPY --chown=algo:algo {{config.local_dependency_src_path}} {{config.local_dependency_dest_path}}
 {% endif %}
 
 ENV HOME=/home/algo
+{% if config.local_dependency_src_path is defined %}
```

Comment:
Using the `local_dependency_src_path` property for build environment detection since @zeryx uses this for his local IPA dev/test workflow. In the case where we're building locally, we won't have `customize-container.sh` available, nor will any custom CA certs be available.

If there's a better way to detect or improve this flow, we can update in the future.

Reply:
This is only necessary when building new language packages so that you don't have to publish all of your intermediate changes to a central repository. Honestly going forwards I think we could remove this, however everything LGTM